[supervisord]
nodaemon=true
user=root
logfile=/var/log/pidanos/supervisord.log
pidfile=/var/run/supervisord.pid
loglevel=info

[program:pidanos-dns]
command=python -m src.dns_server
directory=/opt/pidanos
user=pidanos
autostart=true
autorestart=true
startretries=3
stdout_logfile=/var/log/pidanos/dns.log
stderr_logfile=/var/log/pidanos/dns.error.log
environment=PYTHONPATH="/opt/pidanos",PIDANOS_CONFIG="/etc/pidanos/pidanos.conf"
priority=10

[program:pidanos-api]
command=python -m src.api_server
directory=/opt/pidanos
user=pidanos
autostart=true
autorestart=true
startretries=3
stdout_logfile=/var/log/pidanos/api.log
stderr_logfile=/var/log/pidanos/api.error.log
environment=PYTHONPATH="/opt/pidanos",PIDANOS_CONFIG="/etc/pidanos/pidanos.conf"
priority=20

[program:pidanos-web]
command=python -m src.web_server
directory=/opt/pidanos
user=pidanos
autostart=true
autorestart=true
startretries=3
stdout_logfile=/var/log/pidanos/web.log
stderr_logfile=/var/log/pidanos/web.error.log
environment=PYTHONPATH="/opt/pidanos",PIDANOS_CONFIG="/etc/pidanos/pidanos.conf"
priority=30

[program:pidanos-updater]
command=python -m scripts.update_scheduler
directory=/opt/pidanos
user=pidanos
autostart=true
autorestart=true
startretries=3
stdout_logfile=/var/log/pidanos/updater.log
stderr_logfile=/var/log/pidanos/updater.error.log
environment=PYTHONPATH="/opt/pidanos",PIDANOS_CONFIG="/etc/pidanos/pidanos.conf"
priority=40

[group:pidanos]
programs=pidanos-dns,pidanos-api,pidanos-web,pidanos-updater
priority=999

[unix_http_server]
file=/var/run/supervisor.sock
chmod=0700

[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[supervisorctl]
serverurl=unix:///var/run/supervisor.sock